### DISCUSSION TITLE
enhance translations at crowdin with screenshots

### DISCUSSION THREAD

**Discussion by tolot27** [NightscoutFoundation/xDrip#3379] [Ideas]
Crowdin provides an [Android SDK](https://support.crowdin.com/sdk-android) which allows different levels of integration and support for translators and users:

- Over-The-Air Content Delivery – allows delivering new translations from the Crowdin project to xDrip immediately without new rebuilds
- Real-Time Preview – all the translations that are done in the Editor can be shown in your version of the application in real-time. View the translations already made and the ones you’re currently typing in
- Screenshots – all screenshots [made](https://support.crowdin.com/adding-screenshots) in the application may be automatically sent to your Crowdin project with tagged source strings

The biggest advantage: Automatic Screenshots and Real-Time Preview build for the development team and translators team.

Crowdin does not store any Personally Identifiable Information (PII) about the end-user but xDrip can integrate and additional opt-out option.

**Comment by sulkaharo** [comment]
The OTA delivery sounds like it should be considered with care, as it'll allow Crowdin to track xDrip usage and add logic and a networked dependency to the live app, which would by definition increase probability of users experiencing issues with the live app.

**Comment by tolot27** [comment]
I'm working on this issue and it works so far with the exception that it requires migration to AndroidX. I did this in my fork and I can upload screenshots. But the strings are not tagged yet. Investigating it....

**Comment by tolot27** [comment]
With the new years blog of Crowdin they announced the [Appetize App](https://store.crowdin.com/products/appetize). This is a much more elegant way.  @jamorham Please can you install the Appetize app? I can provide the apk (requires the use Crowdin SDK ... and AndroidX).

**Comment by nightscout-ricsi** [comment]
That would be great help for the translators, because without the context many times I did not have any clue, what that string might mean.

**Comment by tolot27** [comment]
> The OTA delivery sounds like it should be considered with care, as it'll allow Crowdin to track xDrip usage and add logic and a networked dependency to the live app, which would by definition increase probability of users experiencing issues with the live app.

That would just be enabled in a special build for translators and not the general app.

**Comment by Navid200** [comment]
@tolot27 Is this something that needs to change in xDrip or something that needs to change in Crowdin?

**Comment by tolot27** [comment]
I've tried creating screenshots for use in Crowdin. Unfortunately, xDrip still use a very old target API level and you cannot even get previous in the Android Studio design preview. I've already compiled a new xDrip version after migration to AndroidX, which is also used by the Crowdin SDK. This app can be used with the Appetize App but the later must be enabled at Crowdin which can be done only by @jamorham. I just have the managing role.
Hence, I'm waiting for jamorham to to install it.

**Comment by Navid200** [comment]
Thanks for explaining.  
So, there isn't an issue tracking mechanism inside Crowdin?

**Comment by tolot27** [comment]
> So, there isn't an issue tracking mechanism inside Crowdin?

Hmm, not like here at GitHub. It is possible to comment on a string and flag it as an issue. But I don't understand what this has to do with this issue. Or do you just wan't to know how Crowdin works?

**Comment by Navid200** [comment]
If there is an issue tracking mechanism inside Crowdin, considering this is a Crowdin issue, and the fix has to be implemented in Crowdin not in the xDrip code itself, I think it should be in there not in the xDrip set of issues.

On the other hand, if there is no way to track Crowdin issues in Crowdin, then, I guess we have to have it here.

There are too many open issues here and as a result the developers are not looking at them.  This needs to change.  I want to close any open issue that absolutely doesn't have to be open here for that reason.

**Comment by tolot27** [comment]
This issue is not a Crowdin issue, it is an xDrip issue in xDrip code (it has to incorporate the Crowdin SDK which is already done by me) and of the xDrip maintainers. Crowdin cannot contribute anything to solve this issue.
